Sources/SwiftInspectorVisitors/Tests/AssociatedtypeVisitorSpec.swift

```diff
@@ -0,0 +1,136 @@
+        }
+
+        it("finds the associatedtype Iterator") {
+          expect(associatedTypeNameToInfoMap?["Iterator"]).toNot(beNil())
+        }
+
+        it("finds the associatedtype Iterator's initializer") {
+          expect(associatedTypeNameToInfoMap?["Iterator"]?.initializer?.asSource) == "IndexingIterator<Self>"
+        }
+
+        it("finds the associatedtype SubSequence") {
+          expect(associatedTypeNameToInfoMap?["SubSequence"]).toNot(beNil())
+        }
+
+        it("finds the associatedtype SubSequence's inheritance") {
+          expect(associatedTypeNameToInfoMap?["SubSequence"]?.inheritsFromTypes.count) == 1
+          expect(associatedTypeNameToInfoMap?["SubSequence"]?.inheritsFromTypes.first?.asSource) == "Collection"
+        }
+
+        it("finds the associatedtype SubSequence's generic requirements") {
+          expect(associatedTypeNameToInfoMap?["SubSequence"]?.genericRequirements.count) == 2
+          expect(associatedTypeNameToInfoMap?["SubSequence"]?.genericRequirements.first?.leftType.asSource) == "Self.Element"
+          expect(associatedTypeNameToInfoMap?["SubSequence"]?.genericRequirements.first?.relationship) == .equals
+          expect(associatedTypeNameToInfoMap?["SubSequence"]?.genericRequirements.first?.rightType.asSource) == "Self.SubSequence.Element"
+          expect(associatedTypeNameToInfoMap?["SubSequence"]?.genericRequirements.last?.leftType.asSource) == "Self.SubSequence"
+          expect(associatedTypeNameToInfoMap?["SubSequence"]?.genericRequirements.last?.relationship) == .equals
+          expect(associatedTypeNameToInfoMap?["SubSequence"]?.genericRequirements.last?.rightType.asSource) == "Self.SubSequence.SubSequence"
+        }
+
+        it("finds the associatedtype Indices") {
+          expect(associatedTypeNameToInfoMap?["Indices"]).toNot(beNil())
+        }
+
+        it("finds the associatedtype Indices's inheritance") {
+          expect(associatedTypeNameToInfoMap?["Indices"]?.inheritsFromTypes.count) == 1
+          expect(associatedTypeNameToInfoMap?["Indices"]?.inheritsFromTypes.first?.asSource) == "Collection"
+        }
+
+        it("finds the associatedtype Indices's inheritance") {
+          expect(associatedTypeNameToInfoMap?["Indices"]?.inheritsFromTypes.count) == 1
+          expect(associatedTypeNameToInfoMap?["Indices"]?.initializer?.asSource) == "DefaultIndices<Self>"
+        }
+
+        it("finds the associatedtype Indices's generic requirement") {
+          expect(associatedTypeNameToInfoMap?["Indices"]?.genericRequirements.count) == 1
+          expect(associatedTypeNameToInfoMap?["Indices"]?.genericRequirements.first?.leftType.asSource) == "Self.Indices"
+          expect(associatedTypeNameToInfoMap?["Indices"]?.genericRequirements.first?.relationship) == .equals
+          expect(associatedTypeNameToInfoMap?["Indices"]?.genericRequirements.first?.rightType.asSource) == "Self.Indices.SubSequence"
```

Comment:
I usually don't love having multiple assertions per test but this sounds ok to me, since we're basically testing the same _thing_ but just checking all the pieces of it, so I assume when one assertion fails, the others will likely fail too.

Reply:
Exactly. Though the count and the check of the count could theoretically fail at different times... it's def checking multiple aspects of the same thing